enhancements/monitoring/metrics-collection-profiles.md

```diff
@@ -0,0 +1,438 @@
+    regex: "federate_samples|federate_filtered_samples"
+ ```
+
+Note: 
+- the `metricRelabeling` section only keeping two metrics, while the rest is
+dropped.
+- the metrics in the `keep` section were obtained with the help of a script that
+  parsed all Alerts, PrometheusRules and Console dashboards to determine what
+  metrics were actually being used.
+
+Finally, a team that adopts the metrics collection profile feature should also 
+add themselves and their component to the list under 
+[Infrastruture needed](#infrastructure-needed-optional)
+
+#### Variation [optional]
+
+NA
+
+### API Extensions
+
+- Addition of a new field to the `cluster-monitoring-config` ConfigMap
+
+```go
+type PrometheusK8sConfig struct {
+	// Defines the metrics collection profile that Prometheus uses to collect
+	// metrics from the platform components. Supported values are `full` or
+	// `minimal`. In the `full` profile (default), Prometheus collects all
+	// metrics that are exposed by the platform components. In the `minimal`
+	// profile, Prometheus only collects metrics necessary for the default
+	// platform alerts, recording rules, telemetry and console dashboards.
+	CollectionProfile CollectionProfile `json:"collectionProfile,omitempty"`
+}
+```
+
+### Implementation Details/Notes/Constraints
+
+Each OpenShift team that wants to adopt this feature will be responsible for
+providing the different monitors. This work is not trivial. Dependencies between
+operators and their metrics exist. This makes it difficult for developers to
+determine whether a given metric can be excluded from the `minimal` profile or
+not. To aid teams with this effort the monitoring team will provide:
+- a CLI tool that consumes a ServiceMonitor and a list of metrics and
+  generates the `minimal` ServiceMonitor for that operator.
+- an origin/CI test that validates for all Alerts and PrometheusRules that the
+  metrics used by them are present in the `keep` expression of the
+  monitor for the `minimal` profile
+
+
+### Risks and Mitigations
+
+- How are monitors supposed to be kept up to date? In 4.12 a metric that wasn't
+  being used in an alert is now required, how does the monitor responsible for
+  that metric gets updated?
+  - The origin/CI test mentioned in the previous section will fail if there is a
+    resource (Alerts/PrometheusRules/Dashboards) using a metric which is not
+    present in the monitor in question;
+
+- What happens if a user provides an invalid value for a metrics collection profile?
+  - CMO will reconcile and validate that the value supplied is invalid and it
+    will report Degraded=False and fail reconciliation.
+
+- Should we add future profiles? How would we validate such profiles?
+  - Our current validation strategy with only two profiles is quite linear,
+    however, things start becoming more complex and hard to mainain as we
+    introduce new profiles to the mix. 
+  - Some of the things to consider if new profiles are introduce are:
```

Comment:
Could it work if we only introduce/support the new profile for techpreview clusters and GA when all components have been modified to support it?

Reply:
From my POV that would certainly solve the problem 

> How would we ensure teams that adopted metrics collection profiles
        implement the new profile?


But it's still unclear how could we validate the new profile. Regardless I can add that idea to this section